### PR TITLE
refactor w/ arb-ts-v2

### PR DESCRIPTION
**Pull request by DZGoldman**
Includes multiple L1ToL2Messages for a given l1 txn support (including L1ToL2Messages across multiple arb chains). 
Someone should make it look nice

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -26,334 +114,251 @@ export interface RetryableTxs {
...
+          showRedeemButton: false,
+          ...stuffTheyAllHave
+        };
+      }
+      const text = (() => {
+        if (!autoRedeemRec) {
+          return "Auto-redeem not attempted; you can redeem it now:";
+        }
         switch (BigNumber.from(autoRedeemRec.returnCode).toNumber()) {
           case 1:
```
[the remark this answers is not in the document]
Comment (yahgwai) [reply src/App.tsx:250]: I've done this now

File: src/App.tsx [hunk 2]
```diff
@@ -26,334 +114,251 @@ export interface RetryableTxs {
...
 
-      try {
-        const autoRedeemRec = await getArbTransactionReceipt(
-          l2Provider,
-          autoRedeemHash
-        );
+      const l2Provider = l1ToL2Message.l2Provider as JsonRpcProvider;
+      const autoRedeemRec = await getRawArbTransactionReceipt(
+        l2Provider,
+        l1ToL2Message.autoRedeemId
```
[the remark this answers is not in the document]
Comment (yahgwai) [reply src/App.tsx:232]: I'll add this to our misc doc

**Review by yahgwai — APPROVED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -26,334 +114,251 @@ export interface RetryableTxs {
...
+      );
 
+      // sanity check; should never occur
+      if (autoRedeemRec && autoRedeemRec.status === "0x1") {
```
Comment (yahgwai) [inline src/App.tsx:236]: I think I've updated status to be an number now - like a normal eth tx

**Comment by DZGoldman** [comment]
Looks like all the v2 changes proposed here are on https://github.com/OffchainLabs/arbitrum/pull/2002; gonna merge this as this, then we can bump the version / update accordingly

**Pull request merged**